### PR TITLE
Feat: location buttons

### PR DESCRIPTION
**Pull request by Tristan-C789**
This pull request includes several changes to improve the navigation and location handling in the `home` and `map` components of the app. The most important changes include updating the `Challenge` component to handle location data, modifying the `MapScreenViewModel` to use `GeoPoint` for location, and adding navigation to the `MapScreen` and `MaximizeScreen` components.

### Navigation and location handling improvements:

* [`app/components/home/Challenge.tsx`](diffhunk://#diff-1de67920e270ee2b4e052079c38a6d69b783524b8be0eb8c55237e89c6f292efR172-R186): Added a condition to check for `challengeDB.location` and navigate to `MapScreen` with the appropriate parameters when the location button is pressed.
* [`app/src/views/home/maximize_screen.tsx`](diffhunk://#diff-0176091e71b2795993b4d1a6d7735a3e1d94e93764fbd5e224b60fe4169a824aR86-R101): Added a location button that navigates to `MapScreen` with the challenge's location and other necessary parameters.
* [`app/src/views/map/map_screen.tsx`](diffhunk://#diff-19dc38a502ea7299c284084d3ba1e097d75568407b94f7145689a84ec9ecef04R21-R30): Modified to accept `route` as a parameter and pass `firstLocation` to `useMapScreenViewModel`. Added navigation to `Maximize` screen when a map callout is pressed.

### Location data handling improvements:

* [`app/src/viewmodels/map/MapScreenViewModel.tsx`](diffhunk://#diff-84004b0c03c2c33123913e725bedbb5bc12d95f2d85f0e5ec27babb525919689L5-R14): Changed the location handling to use `GeoPoint` instead of `LocationObject` for better integration with Firestore. Updated the default location and user location state initialization. 

### Minor fixes:

* [`app/src/viewmodels/home/MaximizeScreenViewModel.tsx`](diffhunk://#diff-1c48093449c1a5da8973bd045d25909079b30ab6ff48de0065936d3cbed1f0f5L32-R32): Fixed the `navigateGoBack` function to properly call `navigation.goBack()`.
* [`app/components/home/Challenge.tsx`](diffhunk://#diff-1de67920e270ee2b4e052079c38a6d69b783524b8be0eb8c55237e89c6f292efL75-R75): Removed the unnecessary dependency array in the `useEffect` hook.

### CODE REVIEW

File: app/src/viewmodels/home/MaximizeScreenViewModel.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export function useMaximizeScreenViewModel(
     firestoreCtrl
       .getCommentsOf(challenge.challenge_id ?? "")
       .then((comments) => {
-        const sortedComments = comments.sort(
+        const sortedComments = comments.toSorted(
```
Comment (margdsq) [inline app/src/viewmodels/home/MaximizeScreenViewModel.tsx:46]: Good idea ! Thank you

File: app/src/views/home/maximize_screen.tsx [hunk 2]
```diff
@@ -83,6 +83,22 @@ export default function MaximizeScreen({
           </ThemedView>
         </ThemedView>
```
Comment (margdsq) [inline app/src/views/home/maximize_screen.tsx:85]: This is good thank you, good idea to mask the location button if there isn't any

File: app/src/viewmodels/map/MapScreenViewModel.tsx [hunk 3]
```diff
@@ -26,21 +21,23 @@ const defaultLocation = {
 export function useMapScreenViewModel(
   firestoreCtrl: FirestoreCtrl,
   navigation: any,
+  firstLocation: GeoPoint | undefined,
 ) {
   const [permission, setPermission] = useState<boolean>(false);
-  const [userLocation, setUserLocation] = useState<LocationObject | undefined>(
-    undefined,
+  const [userLocation, setUserLocation] = useState<GeoPoint | undefined>(
+    firstLocation,
```
Comment (margdsq) [inline app/src/viewmodels/map/MapScreenViewModel.tsx:28]: What is 'firstLocation' ? Is it default location ?

**Review by margdsq — CHANGES_REQUESTED** [review]

File: app/src/viewmodels/map/MapScreenViewModel.tsx (hunk 3 above)
Comment (Tristan-C789) [reply app/src/viewmodels/map/MapScreenViewModel.tsx:28]: Well that's quite a tricky vocabulary issue : the `defaultLocation` variable is the location that is used when no location is given in the parameters of the screen AND the user refuses to give their location. The `firstLocation` is the location that *may* be given through the screen parameters. I agree the naming is not very clear, do you have any suggestions ? There are already comments explaining this above the variable declaration.

Comment (margdsq) [reply app/src/viewmodels/map/MapScreenViewModel.tsx:28]: OK I see ! It's fine then

**Review by margdsq — APPROVED** [review]
looks good ! The test of the CI don't pass but the issue is resolved by another PR and it had nothing to do with your PR. Good job !

**Pull request merged**